frontend/src/util/i18n.js

```diff
@@ -0,0 +1,121 @@
+const galleryEnTranslation = require('../../dist/app/gallery_en_translation.json')
+const galleryFrTranslation = require('../../dist/app/gallery_fr_translation.json')
+const galleryPtTranslation = require('../../dist/app/gallery_pt_translation.json')
+
+export const getBrowserLang = () => {
+  const browserLang = navigator.language
+
+  if (['en', 'fr', 'pt'].includes(browserLang)) return browserLang
+  if (browserLang.includes('fr')) return 'fr' // for fr-XX
+  if (browserLang.includes('pt')) return 'pt' // for pt-XX
+
+  return 'en'
+}
+
+i18n
+  .use(reactI18nextModule)
+  .init({
+    fallbackLng: getBrowserLang(),
+    // have a common namespace used around the full app
+    ns: ['translation'], // namespace
+    defaultNS: 'translation',
+    nsSeparator: false,
+    keySeparator: false,
+    debug: false,
+    resources: {
+      en: {
+        translation: {
+          ...enTranslation, // frontend_lib
+          ...en, // frontend
+          ...htmlDocEnTranslation, // html-document
+          ...threadEnTranslation, // thread
+          ...fileEnTranslation, // file
+          ...folderEnTranslation, // folder
+          ...wsAdvancedEnTranslation, // advanced workspace
+          ...adminWsUserEnTranslation, // admin workspace user
+          ...wsEnTranslation, // workspace
+          ...agendaEnTranslation, // agenda
+          ...collaborativeDocumentEditionEnTranslation,
+          ...shareFolderEnTranslation, // share folder,
+          ...galleryEnTranslation // gallery
+        }
+      },
+      fr: {
+        translation: {
+          ...frTranslation, // frontend_lib
+          ...fr, // frontend
+          ...htmlDocFrTranslation, // html-document
+          ...threadFrTranslation, // thread
+          ...fileFrTranslation, // file
+          ...folderFrTranslation, // folder
+          ...wsAdvancedFrTranslation, // advanced workspace
+          ...adminWsUserFrTranslation, // admin workspace user
+          ...wsFrTranslation, // workspace
+          ...agendaFrTranslation, // agenda
+          ...collaborativeDocumentEditionFrTranslation,
+          ...shareFolderFrTranslation, // share folder
+          ...galleryFrTranslation // gallery
+        }
+      },
+      pt: {
+        translation: {
+          ...ptTranslation, // frontend_lib
+          ...pt, // frontend
+          ...htmlDocPtTranslation, // html-document
+          ...threadPtTranslation, // thread
+          ...filePtTranslation, // file
+          ...folderPtTranslation, // folder
+          ...wsAdvancedPtTranslation, // advanced workspace
+          ...adminWsUserPtTranslation, // admin workspace user
+          ...wsPtTranslation, // workspace
+          ...agendaPtTranslation, // agenda
+          ...collaborativeDocumentEditionPtTranslation,
+          ...shareFolderPtTranslation, // share folder
+          ...galleryPtTranslation // gallery
+        }
+      }
```

Comment:
same thing here, I think it would be better to use a loop to fill these, using some structure like `translationsByAppAndLanguage['gallery']['fr']` for instance, or even directly an object that uses the structure of the `resources` field.

Reply:
I think you're right, but I wont do it in this PR